core/src/array/array_sorted_read_state.cc

```diff
@@ -2437,30 +2088,57 @@ Status ArraySortedReadState::read_dense_sorted_col() {
     return array_->read_default(
         copy_state_.buffers_, copy_state_.buffer_sizes_);
 
-  // Iterate over each tile slab
+  // Handle resume copy
+  if (resume_copy_) {
```

Comment:
You can remove all the redundant code here by consolidating the read algorithm to one function and then passing in the `next_tile` and `copy_tile` methods as arguments.  That would get rid of (read_dense_sorted_row, read_sparse_sorted_row, etc.) and make it easier to extend to new layouts.